From: Aron Xu <aron@debian.org>
Date: Sat, 24 Jan 2015 09:47:25 +0800
Subject: sync call method of str_init()

---
 src/main-ctl-dbus.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main-ctl-dbus.c b/src/main-ctl-dbus.c
index 537ead8..61bc084 100644
--- a/src/main-ctl-dbus.c
+++ b/src/main-ctl-dbus.c
@@ -946,7 +946,7 @@ static void method_introspect(main_server_st * s, struct dbus_ctx *ctx,
 
 	mslog(s, NULL, LOG_DEBUG, "ctl: introspect");
 
-	str_init(&buf);
+	str_init(&buf, ctx);
 
 	ret = str_append_data(&buf, XML_HEAD, sizeof(XML_HEAD) - 1);
 	if (ret < 0) {
